src/inmanta/data/__init__.py

```diff
@@ -2194,6 +2295,153 @@ async def get_logs_for_version(
+        :return:
+        """
+        for o in order.split(" "):
+            possible = ["ASC", "DESC", "NULLS", "FIRST", "LAST"]
+            if o not in possible:
+                raise RuntimeError(f"The following order can not be applied: {order}, {o} should be one of {possible}")
+
+        valid_field_names = list(cls._fields.keys())
+        valid_field_names.extend(["timestamp", "level", "msg"])
+        if order_by_column not in valid_field_names:
+            raise RuntimeError(f"{order_by_column} is not a valid field name.")
+
+        return ColumnNameStr(order_by_column), OrderStr(order)
+
+    @classmethod
+    def _get_resource_logs_base_query(
+        cls,
+        select_clause: str,
+        environment: uuid.UUID,
+        resource_id: m.ResourceIdStr,
+        offset: int,
+    ) -> Tuple[str, List[object]]:
+        query = f"""{select_clause}
+                    FROM
+                    (SELECT action_id, action, (unnested_message ->> 'timestamp')::timestamptz as timestamp,
+                    unnested_message ->> 'level' as level,
+                    unnested_message ->> 'msg' as msg,
+                    unnested_message
+                    FROM {cls.table_name()}, unnest(resource_version_ids) rvid, unnest(messages) unnested_message
+                    WHERE environment = ${offset} AND rvid LIKE ${offset + 1}) unnested
+                    """
+        values = [cls._get_value(environment), cls._get_value(f"{resource_id}%")]
+        return query, values
+
+    @classmethod
+    async def get_logs_paged(
+        cls,
+        database_order: DatabaseOrder,
+        limit: int,
+        environment: uuid.UUID,
+        resource_id: m.ResourceIdStr,
+        start: Optional[Any] = None,
+        end: Optional[Any] = None,
+        connection: Optional[asyncpg.connection.Connection] = None,
+        **query: Tuple[QueryType, object],
+    ):
+        order_by_column = database_order.get_order_by_db_column_name()
+        order = database_order.get_order()
+        filter_statements, values = cls._get_list_query_pagination_parameters(
+            database_order=database_order,
+            id_column=ColumnNameStr("resource_version_id"),
+            first_id=None,
+            last_id=None,
+            start=start,
+            end=end,
+            **query,
+        )
+        db_query, base_query_values = cls._get_resource_logs_base_query(
+            select_clause="SELECT action_id, action, timestamp, unnested_message ",
+            environment=environment,
+            resource_id=resource_id,
+            offset=len(values) + 1,
+        )
+        values.extend(base_query_values)
+        if len(filter_statements) > 0:
+            db_query += cls._join_filter_statements(filter_statements)
+        backward_paging = ("ASC" in order and end) or ("DESC" in order and start)
+        if backward_paging:
+            if "ASC" in order:
+                backward_paging_order = order.replace("ASC", "DESC")
```

Comment:
This part I don't understand. The value can only be `ASC` or `DESC` (or `asc` and `desc`?)
Why not simply set it to `DESC` ?

Reply:
or even make this order an enum with an invert function?